notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/3_Forecast.ipynb

```diff
@@ -0,0 +1,972 @@
+    "\n",
+    "Here we choose the DeepAR+ algorithm as it is capable of building a global model based on all the different target time series data. Additonally, like the Prophet and NPTS algorithms, DeepAR+ can also incorporate information from the related time series which we provide here. \n",
+    "\n",
+    "For more details on the DeepAR+ algorithm and the differences between this and DeepAR, please see: https://docs.aws.amazon.com/forecast/latest/dg/aws-forecast-recipe-deeparplus.html\n",
+    "\n",
+    "Currently only the DeepAR+ algorithm supports hyperparamter optimization. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictorName = 'web_traffic_forecast' + 'DeepARPlus'\n",
+    "forecastHorizon = 15\n",
+    "algorithmArn = 'arn:aws:forecast:::algorithm/Deep_AR_Plus' \n",
+    "# choose an algorithm here or set AutoML to be true. Possible algorithmARN choices are:\n",
+    "#ARIMA (no related time series): arn:aws:forecast:::algorithm/ARIMA\n",
+    "#ETS (no related time series) arn:aws:forecast:::algorithm/ETS\n",
+    "#NPTS: arn:aws:forecast:::algorithm/NPTS\n",
+    "#Prophet: arn:aws:forecast:::algorithm/Prophet\n",
+    " "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Predictor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    create_predictor_response=forecast.create_predictor(PredictorName=predictorName, \n",
+    "                                                  ForecastHorizon=forecastHorizon,\n",
+    "                                                  AlgorithmArn=algorithmArn,\n",
+    "                                                  PerformAutoML=False, # change to true if want to perform AutoML\n",
+    "                                                  PerformHPO=False, # change to true to perform HPO\n",
+    "                                                  EvaluationParameters= {\"NumberOfBacktestWindows\": 1, \n",
+    "                                                                         \"BackTestWindowOffset\": 15}, \n",
+    "                                                  InputDataConfig= {\"DatasetGroupArn\": datasetGroupArn},\n",
+    "                                                  FeaturizationConfig= {\"ForecastFrequency\": \"D\", \n",
+    "                                                                        }\n",
+    "                                                 )\n",
+    "    predictorArn=create_predictor_response['PredictorArn']\n",
+    "except Exception as e:\n",
```

Comment:
Catching generic "Exception" class to skip creating existing resources, but shouldn't do this as it is possible to catch different types of errors.
Use forecast.exceptions.ResourceAlreadyExistsException specifically.